aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -134,11 +140,15 @@
                 .toString()
             );
         } catch (JSONException | ApiException exception) {
-            onSubscriptionError.accept(new ApiException(
+            // If the subscriptionId was still pending, then we can call the onSubscriptionError
```

Comment:
For us to get here, we know that the subscription was added to the pendingSubscriptionIds set and we have a good connection. These two exceptions can happen either because of some malformed value passed into the `put` calls, or if `createHeadersForSubscription` throws an `ApiException`.

If the subscription was not pending, it's likely because `cancel()` was invoked some time after the connection was successfully established, but before the subscription handshake completed.